### PR TITLE
Add some host_config arguments to DockerContainer

### PR DESCRIPTION
**Pull request by davidssand**
closes https://github.com/PrefectHQ/prefect/issues/7301

### Example
```
DockerContainer(
    image=my_image,
    memswap_limit="1g",
    mem_limit="1g",
    privileged=True,
)
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/infrastructure/docker.py [hunk 1]
```diff
@@ -348,6 +369,9 @@ def _build_container_settings(
             extra_hosts=self._get_extra_hosts(docker_client),
             name=self._get_container_name(),
             volumes=self.volumes,
+            mem_limit=self.mem_limit,
+            memswap_limit=self.memswap_limit,
+            privileged=self.privileged,
```
Comment (davidssand) [inline src/prefect/infrastructure/docker.py:374]: These 3 arguments will be added to the `host_config` at `docker.models.containers._create_container_args`.

File: src/prefect/infrastructure/docker.py [hunk 2]
```diff
@@ -226,6 +231,22 @@ class DockerContainer(Infrastructure):
         default=True,
         description="If set, the output will be streamed from the container to local standard output.",
     )
+    memswap_limit: Union[int, str] = Field(
+        default=None,
+        description="Total memory (memory + swap), -1 to disable swap.",
```
Comment (madkinsz) [inline src/prefect/infrastructure/docker.py:236]: What's the default Docker behavior for this one?

**Review by madkinsz — APPROVED** [review]
Thanks for contributing!

File: src/prefect/infrastructure/docker.py (hunk 2 above)
Comment (davidssand) [reply src/prefect/infrastructure/docker.py]: Docker does not set a `mem_limit` by default, as explained here: https://codefresh.io/blog/docker-memory-usage/#:~:text=Docker%20does%20not%20apply%20memory,consume%20the%20entire%20host's%20memory.

`memswap_limit` only has meaning if `mem_limit` is set, see: https://docs.docker.com/config/containers/resource_constraints/#--memory-swap-details
If `mem_limit` is set, `memswap_limit` defaults to allowing the container to use as much swap as memory.

I've improved the docs of `DockerContainer` to explain this behavior.

**Pull request merged**